pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/ForwardIndexHandler.java

```diff
@@ -72,34 +75,34 @@
  * 1. Change compression on raw SV and MV columns.
  * 2. Enabling dictionary on a raw column.
  * 3. Disable forward index on a column where it is enabled.
+ * 4. Enable forward index on a forward index disabled column
  *
  *  TODO: Add support for the following:
  *  1. Disable dictionary
  *  2. Segment versions < V3
- *  3. Enable forward index on a forward index disabled column
  */
-public class ForwardIndexHandler implements IndexHandler {
+public class ForwardIndexHandler extends BaseForwardIndexBasedIndexHandler {
   private static final Logger LOGGER = LoggerFactory.getLogger(ForwardIndexHandler.class);
 
-  private final SegmentMetadata _segmentMetadata;
-  private final IndexLoadingConfig _indexLoadingConfig;
+  // This should contain a list of all indexes that need to be rewritten if the dictionary is enabled or disabled
+  private static final List<ColumnIndexType> DICTIONARY_RELATED_INDEX_TO_REMOVE =
```

Comment:
Should this be called as DICTIONARY_BASED_INDEX_TO_REWRITE instead ?

If this is both REMOVE and REWRITe, then I think inverted index should also be added to the list. 

Based on Vivek's recent work, inv index is deleted after disabling dictionary at runtime.